installation_policies/xSDK_spack_variant_guidelines.md

```diff
@@ -0,0 +1,88 @@
+
+---
+
+**2. Precision**
+
+This concerns the size of compile-time configured floating-point variables.
+
+We recommend using multi-valued variant `precision`.
+
+`precision`: The multi-valued variant `precision` can accept multiple precision descriptions. The names `single` and `double` are recommended for 32-bit/64-bit respectively. Other options may include `half` for 16-bit, `extended` for 80-bit or `quad` for 128-bit.
+
+---
+
+**3. Shared/Static libraries**
+
+These options determine how libraries are built.
+
+**_Note_**
+There has been discussion about adopting a multi-value variant for this purpose, which may better cover the possible permutations. However, the following recommendations have been determined based on a review of current usage. It is possible this will change in the future.
+
+We recommend using one or more of following boolean variants which have been adopted widely in Spack:
+
+- `shared`: A boolean variant. If True, then shared libraries will be built.
+- `static`: A boolean variant. If True, then static libraries will be built.
+- `pic`: A boolean variant. If True, then produce position independent code.
+
+The meaning and combination of these values is expected to follow conventions given below (based on CMake):
+
+If `shared` is the only option present, then enabling this option is expected to build shared libraries *instead* of static, and these will be built using _pic_ (position independent code). If False, then static libraries will be built without pic.
+
+If the package author wishes to provide an option to build both static and shared libraries, then use both `static` and `shared` variants, and set both to True.
+
+To build shared libraries but without _pic_, then use `shared` and `pic`, and set `pic` to False.
+
+Permutations of package variants that are unsupported or infeasible should be denoted in a `conflict` statement. E.g. `conflicts('~static~shared')`
+
+---
+
+**4. Debug/build_type**
+
+This option configures debugging symbols and optimization.
+
+We recommend using the multi-valued variant `build_type`.
+
+Note that packages that use the `CMakePackage` class already have this variant set with the default value `RelWithDebInfo` as described in the [Spack CMakePackage documentation](https://spack.readthedocs.io/en/latest/build_systems/cmakepackage.html#cmake-build-type). Package author's may rely on this default if all of the value options are supported. Alternatively the `build_type` variant may be defined in **package.py** to override the default.
+
+`build_type`: A multi-value variant which must include at least `Debug` and `Release` options. Other options, mirroring their CMake counterparts, may include `RelWithDebInfo` (highly recommended) and `MinSizeRel`. A brief description of these options is given below.
```

Comment:
If I understand this correctly all cmake packages already support `build_type` variant.

How about other non-cmake pacakges. Some of them have `release` variant. If I understand this - they should be changed to `build_type`. [as its currently listed as mandatory]

And how does this apply to python package?

Reply:
With the change of the language suggested in the meeting on 10/8, I will approve.